app/src/ui/branches/branches-container.tsx

```diff
@@ -134,9 +152,9 @@ export class BranchesContainer extends React.Component<
             allBranches={this.props.allBranches}
             recentBranches={this.props.recentBranches}
             onItemClick={this.onItemClick}
-            filterText={this.state.filterText}
-            onFilterKeyDown={this.onFilterKeyDown}
-            onFilterTextChanged={this.onFilterTextChanged}
+            filterText={this.state.branchFilterText}
+            onFilterKeyDown={this.closeBranchFoldoutOnEsc}
```

Comment:
I think this handler should be attached to the entire container. You should be able to hit escape while navigating the list of PRs as well as in the filter text.

Reply:
Do you mean attaching it to the `TabBar` component? If so, one of the things I struggled with was adding an event handler to plain Dom node and getting it to respond to React's synthetic events.

Reply:
No I mean we should attach it to the branches-container div. So something like

```tsx
<div className="branches-container" onKeyDown={...}>
```

And the in the handler we'd check whether the event was handled or and if the key was 'escape'. The next change would be to have `TextBox` handle `Escape` when `type=search` and it's not empty. That way we can keep the logic specific to search inputs contained within the text component and everyone else that needs to handle escape can do their thing without having to take search boxes into account. We could take a look at this together next week if you'd like